homeassistant/components/yale_smart_alarm/alarm_control_panel.py

```diff
@@ -75,51 +81,85 @@ async def async_setup_entry(
 class YaleAlarmDevice(CoordinatorEntity, AlarmControlPanelEntity):
     """Represent a Yale Smart Alarm."""
 
+    _attr_code_arm_required = False
+    _attr_supported_features = SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_AWAY
+
     def __init__(self, coordinator: YaleDataUpdateCoordinator) -> None:
         """Initialize the Yale Alarm Device."""
         super().__init__(coordinator)
-        self._attr_name: str = coordinator.entry.data[CONF_NAME]
+        self._coordinator = coordinator
+        self._attr_name = coordinator.entry.data[CONF_NAME]
         self._attr_unique_id = coordinator.entry.entry_id
-        self._identifier: str = coordinator.entry.data[CONF_USERNAME]
-
-    @property
-    def device_info(self) -> DeviceInfo:
-        """Return device information about this entity."""
-        return DeviceInfo(
-            identifiers={(DOMAIN, self._identifier)},
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, coordinator.entry.data[CONF_USERNAME])},
             manufacturer=MANUFACTURER,
             model=MODEL,
-            name=str(self.name),
+            name=self._attr_name,
         )
 
-    @property
-    def state(self):
-        """Return the state of the device."""
-        return STATE_MAP.get(self.coordinator.data["alarm"])
-
-    @property
-    def available(self):
-        """Return if entity is available."""
-        return STATE_MAP.get(self.coordinator.data["alarm"]) is not None
-
-    @property
-    def code_arm_required(self):
-        """Whether the code is required for arm actions."""
-        return False
-
-    @property
-    def supported_features(self) -> int:
-        """Return the list of supported features."""
-        return SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_AWAY
-
-    def alarm_disarm(self, code=None):
+    async def async_alarm_disarm(self, code=None) -> None:
```

Comment:
It feels to me like failing to change the state of the alarm is something as I user I would want feedback about -- e.g. showing an error in the UI when my arm/disarm command failed.

I am all for improving failure handling with better exception catching, but not sure that just logging a warning is the right thing. Would it make sense to throw a HomeAssistantError here for those cases, or do you have additional rationale for why swallowing is the right thing to do that I'm not aware of? (e.g. UI treatment, i'm not familiar with these cards)

Reply:
I would be more than happy to show a good response to the user (if done through the UI) but I have not found a good exception to raise for this purpose. Or you possibly just think I should raise the general HomeAssistantError?

Reply:
I think catching and rethrowing HomeassistantError is how I have seems this handled

Reply:
I will test

Reply:
I went with the suggestion to throw a HomeAssistantError